lms/djangoapps/course_home_api/outline/views.py

```diff
@@ -443,3 +443,34 @@ def save_course_goal(request):
             })
         except Exception:
             raise UnableToSaveCourseGoal
+
+
+@api_view(['POST'])
+def unsubscribe_from_course_goal_by_token(request, token):
+    """
+    API calls to unsubscribe from course goal reminders.
+
+    Note that this does not require authentication - this view may be hit from an email on a different device than
+    normal or whatever. We should still be able to unsubscribe the user. Instead, we use a token in the email to
+    validate that they have permission to unsubscribe.
+
+    This endpoint is very tightly scoped (only unsubscribe: no subscribing, no PII) because it is unauthenticated.
+
+    **Example Requests**
+        POST api/course_home/v1/unsubscribe_from_course_goal/{token}
+
+    **Example Response Data**
+        {'course_title': 'Cats & Dogs In Canadian Media'}
+
+    Returns a 404 response if the token was not found. Otherwise, returns some basic course info. But no PII.
+    """
+    # First update the goal
+    goal = get_object_or_404(CourseGoal, unsubscribe_token=token)
+    goal.subscribed_to_reminders = False
+    goal.save()
+
+    # Now generate a response
+    course_overview = get_course_overview_or_404(goal.course_key)
+    return Response({
+        'course_title': course_overview.display_name,
```

Comment:
... Is this being used? I don't know the full design off the top of my head.

Reply:
Not yet - my plan was for the email to link to something like `learning.edx.org/goal-unsubscribe/<token>` or whatever.

Then the MFE shows a spinner, makes this call to the backend, and displays a message like "Congrats, you've unsubscribed to XXX"

That XXX will be from this json response. (I could have done things different, like have the title be part of the request from the email: `learning.edx.org/goal-unsubscribe/<token>?title=XXX` - but figured I'd keep the unsubscribe URLs as opaque as possible.

Reply:
Has this been validated through Product Design? I don't disagree with your thought process and very likely see the MFE doing something like what you said. Just want to make sure we aren't adding in data we may not use or missing data we know we want. 

Updating it is easy too so this is all pretty nitty, just wanted to comment.

Reply:
Naw this level of detail (title from endpoint vs title encoded in URL) was not specified anywhere. Just an implementation decision by me.

The MFE side of this is all in the [same ticket](https://openedx.atlassian.net/browse/AA-907), so it will definitely be used, because I'm about to turn around and use it on the MFE side.

The only design I have is a single figma screen that has the course title on it. Since the MFE wouldn't want to show the success screen until the endpoint returns with a 200 anyway, I figured we could wait until it returns to have a title in hand.

Plus, you could imagine if the endpoint returned PII (which I caution against in code comments, but play along for a second) - we wouldn't want that encoded in the URL, we'd want to wait for the return. Likewise, I figured maybe just be super secretive until we know the token is valid. And since we don't need the title until the endpoint returns... :shrug:

Reply:
ALSO, per the "old links in emails" thought that had me storing the token long term in the db... If we encode what info the current MFE needs in the URL, and then in six months think we also need some new piece of data, we have a problem. This way, the endpoint can update what info to give back, as the MFE evolves.

Reply:
Ah super sorry. I have no questions about query params vs having the endpoint return the data (and would agree that the endpoint returning the relevant data makes more sense). My question was specifically about the data you're choosing to send (course display name) and if we had designs that planned on using it or if it was you thinking it could be helpful.  I'm all here for the server being the smart one and not the url.

Reply:
Ah gotcha yeah. The design is just the figma linked in that ticket, which only requires the course title right now. :+1: